devtools/src/org/labkey/devtools/ToolsController.java

```diff
@@ -365,4 +378,168 @@ public void addNavTrail(NavTree root)
+            }
+
+            HtmlStringBuilder builder = HtmlStringBuilder.of();
+
+            if (!missingModules.isEmpty())
+            {
+                missingModules.sort(null);
+                builder
+                    .append("The following actions could not be resolved to a module running in this deployment:")
+                    .append(HtmlString.unsafe("<br><br>\n"));
+                missingModules.forEach(id->builder.append(id.toString()).append(HtmlString.unsafe("<br>\n")));
+                builder.append(HtmlString.unsafe("<br>\n"));
+            }
+
+            if (!missingActions.isEmpty())
+            {
+                missingActions.sort(null);
+                builder
+                    .append("The following actions do not exist:")
+                    .append(HtmlString.unsafe("<br><br>\n"));
+                missingActions.forEach(id->builder.append(id.toString()).append(HtmlString.unsafe("<br>\n")));
+            }
+
+            return new HtmlView(builder);
+        }
+
+        private void addActionIds(List<ControllerActionId> actionIds, File file) throws IOException
+        {
+            if (file.isDirectory())
+            {
+                // Crawl all the files in this directory
+                for (File f : file.listFiles(File::isFile))
+                {
+                    addActionIds(actionIds, f);
+                }
+            }
+            else
+            {
+                Pattern pattern = Pattern.compile("ControllerActionId\\(\"(.+?)\", \"(.+?)\"\\)");
+                Matcher matcher = pattern.matcher("");
+
+                String line;
+
+                try (BufferedReader br = Readers.getReader(file))
+                {
+                    while ((line = br.readLine()) != null)
+                    {
+                        matcher.reset(line);
+                        if (matcher.find())
+                            actionIds.add(new ControllerActionId(matcher.group(1), matcher.group(2)));
+                    }
+                }
+            }
+        }
+
+        @Override
+        public void addNavTrail(NavTree root)
+        {
+            addBeginNavTrail(root);
+            root.addChild("Check Crawler Actions");
+        }
+
+        private class ControllerActionId implements Comparable<ControllerActionId>
+        {
+            private final String _controller;
+            private final String _action;
+
+            public ControllerActionId(String controller, String action)
+            {
+                _controller = controller;
+                _action = action;
+            }
+
+            public String getController()
+            {
+                return _controller;
+            }
+
+            public String getAction()
+            {
+                return _action;
+            }
+
+            @Override
+            public String toString()
+            {
+                return "/" + _controller + "/" + _action;
+            }
```

Comment:
Should this use modern `controller-action` style?
```suggestion
            {
                return "/" + _controller + "-" + _action;
            }
```

Reply:
Yes it should!